### PR TITLE
Refactor bookSearch function

### PR DESCRIPTION
**Pull request by Marie-L**
## What is this PR trying to achieve?

- break up `bookSearch` function in accordance with single responsibility principle
- prevent `.innerHTML` being used multiple times

### CODE REVIEW

File: public/js/scripts.js [hunk 1]
```diff
@@ -1,56 +1,93 @@
...
+}
+
+function successResult (data) {
+
+  /* .map method returns a new array,  markUp loops over each book and creates a new array item, the first param must be a function and its param is the array item `book` that will be looped over. Then, .join converts an array into a string (empty string passed through .join allows empty space to be printed).Then call 'printTitle' function to print string. Finally add .innerHTML so that it adds the mark up to the page once.
+  Template literals `${string}` are used to allow embedded expressions.
+   */
+
+
+    const markUp = data.items.map(function(book){
```
Comment (jamesgorrie) [inline public/js/scripts.js]: Can we use ES6?

Comment (jamesgorrie) [reply public/js/scripts.js]: Nice on the map though!

File: public/js/scripts.js [hunk 2]
```diff
@@ -1,56 +1,93 @@
...
+          ${printPublisher(book.volumeInfo.publisher)}
+          ${linkPreview(book.volumeInfo.previewLink)}
+        </div>
+      `
+    }).join('');
+
+    results.innerHTML = markUp;
+}
+
+function printTitle(title) {
```
Comment (jamesgorrie) [inline public/js/scripts.js]: `print` is a word that is often used to say "write out to console". Maybe something descriptive like `getTitleMarkup()` could help disambiguate things.

File: public/js/scripts.js [hunk 3]
```diff
@@ -1,56 +1,93 @@
-function bookSearch(){
-
+function bookSearch () {
     //user input stored in var
     const find = document.getElementById('textName').value;
     document.getElementById('results').innerHTML = "";
+    asyncReq(find);
```
Comment (jamesgorrie) [inline public/js/scripts.js]: Another way to do this without coupling the `asyncReq` and `successResult` could be return the promise from `asyncReq` and pass the result to `successResult `.

e.g. 
```JS
asyncReq(find).then(successResult)
```

File: public/js/scripts.js (hunk 1 above)
Comment (Marie-L) [reply public/js/scripts.js]: Thanks ! ok, will do @jamesgorrie

Comment (Marie-L) [reply public/js/scripts.js]: @jamesgorrie I think so, but when you have time can you please review this again. Thanks

File: public/js/scripts.js [hunk 4]
```diff
@@ -1,56 +1,115 @@
...
+/* - ES6 syntax: function getTitleMarkup() is the same as const getTitleMarkup = () =>
+- Immediately Invoked Function Expressions or IIFE : function() is the same as ()) => */
+/*********************************************************************************************************************/
 
-    //user input stored in var
+const bookSearch = () => {
+    // the users input is stored in const `find`
     const find = document.getElementById('textName').value;
+    // replace the value of div `results` and set it to empty string
     document.getElementById('results').innerHTML = "";
```
Comment (jamesgorrie) [inline public/js/scripts.js]: Maybe as another PR, you could add a loading state?

File: public/js/scripts.js [hunk 5]
```diff
@@ -1,56 +1,115 @@
...
+        xhr.onreadystatechange = (() => {
+          // readyState holds status of XMLHttpRequest object, 4 = request finished and response is ready
+                if (xhr.readyState == 4) {
+                  // 200 = success status code
+                    if (xhr.status == 200) {
+                      // define what happens when promise is fulfilled - resolve = fulfilled (1 of 3 promise states) - responseText returns the text received by the server. resolve(fulfillmentValue)
+                       resolve(xhr.responseText);
+                    } else {
+                      // define what happens when promise is rejected - reject = rejected ( promise state) - the string will be caught in `.catch()` method
+                        reject('request rejected');
```
Comment (jamesgorrie) [inline public/js/scripts.js]: You might want to return the error here too.

File: public/js/scripts.js [hunk 6]
```diff
@@ -1,56 +1,115 @@
...
+        xhr.open ('GET','https://www.googleapis.com/books/v1/volumes?q=' + searchQuery, true);
+        // send request to server
+        xhr.send();
+        }
+    ));
+}
+
+const successResult = (stringifiedData) => {
+// JSON.parse() method parses (taking input from string and returning other data type)a string as JSON
+    const data = JSON.parse(stringifiedData);
```
Comment (jamesgorrie) [inline public/js/scripts.js]: [Bike shedding](https://en.m.wikipedia.org/wiki/Law_of_triviality), but you could error handle here if the JSON wasn't valid.

**Comment by Marie-L** [comment]
@jamesgorrie Thanks so much for your feedback 🙏! I'll look into all of your suggestions and will make changes at a later date. Thanks again 🙂

**Pull request merged**